src/python/pants/backend/jvm/tasks/javadoc_gen.py

```diff
@@ -63,7 +66,16 @@ def link(target):
 
     args.extend(self.args)
 
-    args.extend(sources)
+    javadoc_classpath_file = os.path.join(self.workdir, '{}.classpath'.format(os.path.basename(gendir)))
```

Comment:
These paths are a bit obfuscated... might be clearer to give these files fixed names within the gendir: ie:
```
os.path.join(gendir, 'classpath.atfile.txt')
```

Reply:
The paths come out as 
```
.pants.d/doc/javadoc/src.java.org.pantsbuild.tools.jar.jar.classpath
.pants.d/doc/javadoc/src.java.org.pantsbuild.tools.jar.jar.source.files
```
It seemed better to have separate files for each target instead of one shared file so you could see what each target was generated with and it would not be overwritten.